numba/np/npyimpl.py

```diff
@@ -602,3 +603,39 @@ def imp(desc):
         return imp
     else:
         raise TypeError('unknown dtype descriptor: {}'.format(desc))
+
+
+@overload(np.core.multiarray.normalize_axis_index)
+def normalize_axis_index_ovl(axis, ndim, msg_prefix=None):
+    if not isinstance(axis, (types.Integer, types.IntegerLiteral)):
+        raise TypeError(
+            "normalize_axis_index: 'axis' argument must be an integer")
+    if not isinstance(axis, (types.Integer, types.IntegerLiteral)):
```

Comment:
```suggestion
    if not isinstance(ndim, (types.Integer, types.IntegerLiteral)):
```
also, unit tests did not catch this which suggests they need implementing.